### DISCUSSION TITLE
Queries about GSOC 2024

### DISCUSSION THREAD

**Discussion by nataliyah123** [MDAnalysis/mdanalysis#4387] [GSoC Q&A] [answered]
hi
I just have a few questions about 'mdanalysis' and your participation in the coming GSOC. 

1. Are you going to participate in GSOC in 2024?
2. What skills set  and experience level do you look for in the contributor eg . python, javascript, frameworks etc
3. If a contributor continues to contribute after the internship is over, is there a chance that person can be hired in the future?

**Comment by RMeli** [comment, marked as answer]
Hi @nataliyah123, welcome to MDAnalysis.

1. We are going to apply again this year to GSoC, but whether we are selected or not again depends only on Google, so we will all have to wait until GSoC organisations are announced to know if we are participating again in 2024.
2. You can find the prerequisites on our [GSoC Wiki](https://github.com/MDAnalysis/mdanalysis/wiki/Google-Summer-Of-Code#prerequisites): we usually require basic knowledge and hands-on experience with molecular dynamics simulations, as well as a basic knowledge of Python. The best proposals in previous years usually came from people using MDAnalysis to analyse their molecular dynamics simulations.
3. I'm not sure if GSoC can be called an internship (I think some Universities do recognise it as such), but MDAnalysis is an open source, volunteer-led organisation. So there are usually no paid positions to be filled (there are some paid positions from time to time, depending on grants we might be able to get, but they are the exception and not the norm). People that contribute substantially to the project over a period of time can be elected as core developers (unpaid and volunteering position).

**Reply by nataliyah123** [reply]
@RMeli 
I went through the GSoC wiki, previous discussions etc and have got more questions.?. 
1. Am I late to start off? I think there are people, who are already contributing to the project.
2.  I don't have previous background in molecular dynamics, but I want to learn about it. So, am I eligible to participate?
3. how many contributors are you allowed to take in each round?
4. I have a strong interest in one particular disease progression simulation. But as a newbie, I want to discuss about the feasibility of my idea and whether it will fit in the scope of this project or not (I have no idea whether such functionality is already available or not in your package as I have not gone through the code yet, just watched one or two intro videos). where and who should I reach out to, wiki suggests a mailing list address.

**Reply by RMeli** [reply]
1. Given that we don't even know if we will be selected by Google to participate as an organisation to GSoC 2024, you are definitely not late. ;)
2. If you learn the basics of molecular dynamics you are eligible. We require understanding of MD simulations because there is no MDAnalysis without MD. You can have a look at [The universe as balls and springs: molecular dynamics in Python](https://www.mdanalysis.org/pages/learning_MDAnalysis/#the-universe-as-balls-and-springs-molecular-dynamics-in-python) to start with and look into GROMACS, Amber, and/or OpenMM tutorials next.
3. We usually take 1 to 3 contributors, depending on how many slots Google assigns to us and how many available mentors we have.
4. We are retiring the mailing lists. They are not yet locked, but we highly encourage to use GitHub Discussions. There is a GSoC category which would be appropriate. However, if the project idea is still very much in its infancy and you just want to know if it is in scope, the #GSoC channel in our Discord server might be more appropriate.

**Reply by nataliyah123** [reply]
@RMeli thanks
